scraper.rb

```diff
@@ -47,11 +47,41 @@ class MemberPage < Scraped::HTML
     url.to_s
   end
 
+  field :start_date do
+    date_from(
+      recent_electoral_history
+                  .scan(/(?:Elected on|Oath of Allegiance|casual elections on)[^:]*\:\s*(\d{1,2}\.\d{1,2}\.\d{2,4})/)
+                  .flatten
+                  .last
+    )
+  end
+
+  field :end_date do
+    date_from(
+      recent_electoral_history
+                  .scan(/(?:Resignation from Parliament)[^:]*\:\s*(\d{1,2}\.\d{1,2}\.\d{2,4})/)
+                  .flatten
+                  .last
+    )
+  end
+
   private
 
   def box
     noko.css('div.column2')
   end
+
+  def recent_electoral_history
+    noko.xpath('//tr[contains(.,"Electoral History")]')
+        .text
+        .split('Parliament:')
+        .last
```

Comment:
This tightly binds this class to only working for the most recent term. If we were scraping a list of the members of the Eleventh Term, for example, this would give those memberships the dates of their membership of the Twelfth Term. (Similarly, when the Thirteenth Term begins, the dates of the memberships for the Twelfth Term will be wrong.)